bibliopixel/control/api.py

```diff
@@ -0,0 +1,56 @@
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import urllib
+import getpass, platform, sys, threading
+from .. util import log
+from . control import ExtractedControl
+
+class Api(ExtractedControl):
+    EXTRACTOR = {
+        'keys_by_type': {
+            'direct_command': ['action'],
+            'value_command': ['action', 'value']
+        }
+    }
+
+    def make_value_commands(self, value_command_list):
+        return [{'action': com, 'value': val, 'type': 'value_command'} for (com, val) in value_command_list]
+
+    def make_direct_commands(self, direct_command_list):
+        if direct_command_list == None:
```

Comment:
Better:

    return [{'action': a, 'type': 'direct_command'} for a in direct_command_list or ()]

Reply:
Also, this doesn't use `self` either.